src/core/Akka/Actor/ActorPath.cs

```diff
@@ -676,14 +607,31 @@ public ChildActorPath(ActorPath parentPath, string name, long uid)
             _parent = parentPath;
         }
 
-        /// <summary>
-        /// TBD
-        /// </summary>
+        /// <inheritdoc/>
         public override ActorPath Parent => _parent;
 
-        /// <summary>
-        /// TBD
-        /// </summary>
+        public override IReadOnlyList<string> Elements
+        {
+            get
+            {
+                ActorPath p = this;
+                var acc = new Stack<string>();
+                while (true)
+                {
+                    switch (p)
```

Comment:
what if instead of the switch change it to

```csharp
if (p is RootActorPath _)
   return acc.ToList();
   
acc.Push(p.Name);
p = p.Parent;
```